modules/core/resources/META-INF/smithy/test/pizza.smithy

```diff
@@ -0,0 +1,324 @@
+
+@error("client")
+@httpError(404)
+structure NotFoundError {
+    @required
+    name: String
+}
+
+@error("client")
+structure FallbackError {
+    @required
+    error: String
+}
+
+map Menu {
+    key: String,
+    value: MenuItem
+}
+
+structure AddMenuItemRequest {
+    @httpLabel
+    @required
+    restaurant: String,
+    @httpPayload
+    @required
+    menuItem: MenuItem
+}
+
+structure MenuItem {
+    @required
+    food: Food,
+    @required
+    price: Float
+}
+
+union Food {
+    pizza: Pizza,
+    salad: Salad
+}
+
+structure Salad {
+    @required
+    name: String,
+    @required
+    ingredients: Ingredients
+}
+
+structure Pizza {
+    @required
+    name: String,
+    @required
+    base: PizzaBase,
+    @required
+    toppings: Ingredients
+}
+
+@enum([{name: "CREAM", value: "C"}, {name: "TOMATO", value: "T"}])
+string PizzaBase
+
+@enum([{value: "Mushroom"}, {value: "Cheese"}, {value: "Salad"}, {value: "Tomato"}])
+string Ingredient
+
+list Ingredients {
+    member: Ingredient
+}
+
+@error("server")
+@httpError(502)
+structure GenericServerError {
+    @required
+    message: String
+}
+
+@error("client")
+@httpError(418)
+structure GenericClientError {
+    @required
+    message: String
+}
+
+@readonly
+@http(method: "GET", uri: "/health", code: 200)
+operation Health {
+    input: HealthRequest,
+    output: HealthResponse,
+    errors: [ UnknownServerError ]
+}
+
+structure HealthRequest {
+    @httpQuery("query")
+    @length(min: 0, max: 5)
+    query: String
+}
+
+@freeForm(i : 1, a: 2)
+structure HealthResponse {
+    @required
+    status: String
+}
+
+// This error indicates a fatal, unexpected error has occurred. Fallback strategies ought to be triggered by this
+// error.
+@error("server")
+@httpError(500)
+structure UnknownServerError {
+    @required
+    errorCode: UnknownServerErrorCode,
+
+    description: String,
+
+    stateHash: String
+}
+
+// Define the singular error code that can be returned for an UnknownServerError
+@enum([
+    {
+        value: "server.error",
+        name: "ERROR_CODE"
+    }
+])
+string UnknownServerErrorCode
+
+
+@trait
+document freeForm
+
+
+
+structure GetEnumInput {
+    @required
+    @httpLabel
+    aa: TheEnum
+}
+
+structure GetEnumOutput {
+    result: String
+}
+
+enum TheEnum {
+    V1 = "v1"
+    V2 = "v2"
+}
+
+
+@readonly
+@http(method: "GET", uri: "/get-int-enum/{aa}", code: 200)
+operation GetIntEnum {
+    input := {
+        @required
+        @httpLabel
+        aa: EnumResult
+    }
+    output := {
+        @required
+        result: EnumResult
+    }
+    errors: [ UnknownServerError ]
+}
+
+intEnum EnumResult {
+    FIRST = 1
+    SECOND = 2
+}
+
+@httpRequestTests([
+    {
+        id: "CustomCodeInput"
+        documentation: "test custom code as a label",
+        protocol: simpleRestJson
+        uri : "/custom-code/399"
+        method: "GET"
+        body:""
+        params:{
+            "code": 399
+        }
+    }
+])
+@httpResonsesTests([
```

Comment:
`httpResonsesTests`? seems like this doesn't get automatically validated 🤔

Reply:
apparently not

Reply:
fixed